homework-submission/my-project/src/App.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
+import './App.css';
+import Post from './Post';
+import Data from './Data';
+import Header from './Header';
+
+
+class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      todoData: Data
+    }
+    this.changeHandel = this.changeHandel.bind(this)
+  }
+
+  changeHandel(id) {
+    this.setState(prevState => {
+      const updatedData = prevState.todoData.map(todo => {
+        if (todo.id === id) {
+          todo.done = !todo.done;
+        }
+        return todo
+      })
+      return {
+        todoData: updatedData
+      }
+    })
+  }
+
+  render() {
+    const text = this.state.todoData.length === 0 ? "No item..." : null
+    const newData = this.state.todoData.map((item, key) => <Post key={key} item={item} changeHandel={this.changeHandel} />)
+    return (
+      <div className="App" >
+        <Header/>
+        {text}
+        {newData}
+      </div>
```

Comment:
It is stronger to do it like :
{ this.state.todoData.length === 0 ? "No item..." : newData }

Reply:
how could I use newData before it was defined ?

Reply:
it was defined in line 33, so you can use it.
just replace the code in line 37 and 38 with my code and it will work fine.

Reply:
done!